### PR TITLE
feat: fixed issue with example-vite-react-sdk

### PR DESCRIPTION
**Pull request by okhaimie-dev**
Closes https://github.com/dojoengine/dojo.js/issues/342

## Introduced changes

-

## Checklist

-   [ ] Linked relevant issue
-   [ ] Updated relevant documentation
-   [ ] Added relevant tests
-   [ ] Add a dedicated CI job for new examples
-   [ ] Performed self-review of the code

## Summary by CodeRabbit

- **New Features**
	- Enhanced `dojoConfig` with new properties for improved configuration.
- **Improvements**
	- Updated `useSystemCalls` for better logging and increased timeout for entity state updates.
	- Simplified import path for the JSON manifest in the `dojoConfig` file.
	- Adjusted directory structure for project files during initialization.
- **Bug Fixes**
	- Retained error handling in configuration file updates and system calls.

### CODE REVIEW

File: examples/example-vite-react-sdk/dojoConfig.ts [hunk 1]
```diff
@@ -4,4 +4,10 @@ import manifest from "../../worlds/dojo-starter/manifest_dev.json";
 
 export const dojoConfig = createDojoConfig({
     manifest,
+    rpcUrl: "http://localhost:5050/",
+    toriiUrl: "http://localhost:8080",
+    masterAddress:
+        "0x127fd5f1fe78a71f8bcd1fec63e3fe2f0486b6ecd5c86a0466c3a21fa5cfcec",
+    masterPrivateKey:
+        "0xc5b2fcab997346f3ea1c00b002ecf6f382c5f9c9659a3894eb783c5320f912",
```
[the remark this answers is not in the document]
Comment (okhaimie-dev) [reply examples/example-vite-react-sdk/dojoConfig.ts:14]: fixed

File: examples/example-vite-react-sdk/dojoConfig.ts [hunk 2]
```diff
@@ -4,4 +4,12 @@ import manifest from "../../worlds/dojo-starter/manifest_dev.json";
...
 export const dojoConfig = createDojoConfig({
     manifest,
+    rpcUrl: "http://localhost:5050/",
+    toriiUrl: "http://localhost:8080",
+    masterAddress:
+        "0x127fd5f1fe78a71f8bcd1fec63e3fe2f0486b6ecd5c86a0466c3a21fa5cfcec",
+    // WARNING: Do not use this private key in production!
+    // Use environment variables instead, e.g.: process.env.MASTER_PRIVATE_KEY
+    masterPrivateKey:
+        "0xc5b2fcab997346f3ea1c00b002ecf6f382c5f9c9659a3894eb783c5320f912",
```
Comment (MartianGreed) [inline examples/example-vite-react-sdk/dojoConfig.ts:14]: I think those lines are not required are they ?

**Review by MartianGreed — COMMENTED** [review]
Thank's for contribution, just one detail to clarify but we should be good to go 
 !

File: examples/example-vite-react-sdk/dojoConfig.ts (hunk 2 above)
Comment (okhaimie-dev) [reply examples/example-vite-react-sdk/dojoConfig.ts:14]: I always get a whitescreen without adding those configurations.

These values are used in the `DojoContext`, so I think it is required.

Comment (MartianGreed) [reply examples/example-vite-react-sdk/dojoConfig.ts:14]: I had no issues on my end, just be sure that everything is deployed. 
here are the commands that I ran : 
```bash
katana --dev --dev.no-fee --http.cors_origins '*'
```
in `dojo-starter` directory - be sure that world is deployed on local katana : 

```bash
sozo migrate
```

after command ran successfully you should have world address output in stdio

```bash
torii -w 0x06171ed98331e849d6084bf2b3e3186a7ddf35574dd68cab4691053ee8ab69d7 --http.cors_origins '*' 
```

Other than that, pr is ok, but as those values are default and private keys should be pulled out of dojo config, we shouldn't write them down manually

Let me know if you want to make this clearer

Comment (bb-face) [reply examples/example-vite-react-sdk/dojoConfig.ts:14]: I have no issue as well thanks to the commands above, but once I start using the UI (after creating a burner wallet and selecting it from the dropdown), any commands return the following error: 

```
App.tsx:225 Uncaught (in promise) LibraryError: RPC: starknet_estimateFee with params {
  "request": [
    {
      "type": "INVOKE",
      "sender_address": "0x513866925d3db1da7971a8083107a3da616d444b26d1dd5f3e36675d1d55762",
      "calldata": [
        "0x1",
        "0x1aac0d0befd489010ebd11092e4c3df787ad01287d9b8af337507e54cbd0b41",
        "0x239e4c8fbd11b680d7214cfc26d1780d5c099453f0832beb15fd040aebd4ebb",
        "0x1",
        "0x3"
      ],
      "version": "0x100000000000000000000000000000001",
      "signature": [],
      "nonce": "0x1",
      "max_fee": "0x0"
    }
  ],
  "block_id": "pending",
  "simulation_flags": [
    "SKIP_VALIDATE"
  ]
}
 
        41: Transaction execution error: {"execution_error":"Transaction reverted: Transaction execution has failed:\n0: Error in the called contract (contract address: 0x0513866925d3db1da7971a8083107a3da616d444b26d1dd5f3e36675d1d55762, class hash: 0x07dc7899aa655b0aae51eadff6d801a58e97dd99cf4666ee59e704249e51adf2, selector: 0x015d40a3d6ca2ac30f4031e42be28da9b056fef9bb7357ac5e85627ee876e5ad):\nError at pc=0:4211:\nCairo traceback (most recent call last):\nUnknown location (pc=0:290)\nUnknown location (pc=0:2951)\n\n1: Error in the called contract (contract address: 0x01aac0d0befd489010ebd11092e4c3df787ad01287d9b8af337507e54cbd0b41, class hash: 0x0000000000000000000000000000000000000000000000000000000000000000, selector: 0x0239e4c8fbd11b680d7214cfc26d1780d5c099453f0832beb15fd040aebd4ebb):\nRequested contract address 0x01aac0d0befd489010ebd11092e4c3df787ad01287d9b8af337507e54cbd0b41 is not deployed.\n","transaction_index":0}
    at RpcChannel2.errorHandler (chunk-MQOJJ254.js?v=e82f18f5:17768:13)
    at RpcChannel2.fetchEndpoint (chunk-MQOJJ254.js?v=e82f18f5:17785:12)
    at async Account.getSuggestedFee (chunk-MQOJJ254.js?v=e82f18f5:20313:23)
    at async Account.getUniversalSuggestedFee (chunk-MQOJJ254.js?v=e82f18f5:20302:35)
    at async Account.execute (chunk-MQOJJ254.js?v=e82f18f5:20067:22)
    at async DojoProvider.execute (chunk-X4NSLGG4.js?v=e82f18f5:4234:14)
    at async Object.move (contracts.gen.ts:35:24)
    at async onClick (App.tsx:221:41)
```

Is there anything else needed to run the template?

**Pull request closed without merging**